daisy_workflows/build-publish/rhui/rhua_sync_status.py

```diff
@@ -0,0 +1,154 @@
+#  Copyright 2022 Google Inc. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http:#www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Publish metrics for RHUA sync status."""
+
+import os
+import subprocess
+import time
+
+import requests
+
+
+REPODIR = '/var/lib/rhui/remote_share/symlinks/pulp/content/content'
+
+
+def GetRepodataList(dirname):
+  # Use "find" here because os.walk takes ~30 minutes,
+  # while find takes 10-30 seconds.
+  cmd = ['find', dirname, '-name', 'repodata']
+  output = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  return output.stdout.split()
+
+
+def PublishMetric(metric_type, points, access_token, mig_name):
+  region = GetRegion()
+  url = ('https://monitoring.googleapis.com/v3/projects/'
+         'google.com%3Arhel-infra/timeSeries')
+  headers = {'Authorization': 'Bearer {}'.format(access_token)}
+  metric = {'type': metric_type}
+  resource_labels = {'project_id': '155767908850',
+                     'location': region,
+                     'namespace': 'rhua-sync-status',
+                     'node_id': mig_name}
+  resource = {'type': 'generic_node',
+              'labels': resource_labels}
+
+  time_series = [{'metric': metric,
+                  'resource': resource,
+                  'points': points}]
+
+  data = {'timeSeries': time_series}
+  print(data)
+  r = requests.post(url, json=data, headers=headers)
+  print(r.text)
+
+
+def PublishRepoAge(seconds_since_update, timestamp, access_token, mig_name):
+  print(seconds_since_update)
+  end_time = time.strftime('%Y-%m-%dT%H:%M:%S-00:00', time.gmtime(timestamp))
+  metric_type = 'custom.googleapis.com/rhua_sync_age'
+
+  points = [{'interval': {'endTime': end_time},
+             'value': {'int64Value': seconds_since_update}}]
+
+  PublishMetric(metric_type, points, access_token, mig_name)
+
+
+def PublishRecentUpdates(recent_updates, access_token, mig_name):
```

Comment:
could you add some function comments indicating what each one actually publishes?

Reply:
Done.